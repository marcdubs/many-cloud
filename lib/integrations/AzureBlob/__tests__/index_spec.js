const integration = require("../");
const fs = require("fs");

describe("Azure Blob", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  describe("with provided account and accountKey", () => {
    const expectedAccount = "expected-account";
    const expectedAccountKey = "expected-account-key";
    const expectedURL = "https://not-microsoft.com/";
    const expectedContainer = "myContainer";
    const expectedContainerURL = `${expectedURL}${expectedContainer}`;

    let connection;

    beforeEach(async () => {
      connection = await integration({
        account: expectedAccount,
        accountKey: expectedAccountKey,
        containerURL: expectedContainerURL
      });
    });

    it("sets url", () => {
      expect(connection.containerClient.url).toEqual(expectedContainerURL);
    });

    it("sets accountName", () => {
      expect(connection.containerClient.credential.accountName).toEqual(expectedAccount);
    });
  });

  describe("when not provided explicit settings", () => {
    const expectedAccount = "expected-account";
    const expectedAccountKey = "expected-account-key";
    const expectedURL = "https://not-microsoft.com/";
    const expectedContainer = "myContainer";
    const expectedContainerURL = `${expectedURL}${expectedContainer}`;

    let connection;

    beforeEach(async () => {
      connection = await integration({
        account: expectedAccount,
        accountKey: expectedAccountKey,
        containerURL: expectedContainerURL
      });
    });

    it("sets them to their default values", () => {
      expect(connection.settings.autoMkdirOnFileUpload).toEqual(true);
    });
  });

  describe("when provided with explicit settings", () => {
    const expectedAccount = "expected-account";
    const expectedAccountKey = "expected-account-key";
    const expectedURL = "https://not-microsoft.com/";
    const expectedContainer = "myContainer";
    const expectedContainerURL = `${expectedURL}${expectedContainer}`;

    let connection;

    beforeEach(async () => {
      connection = await integration({
        account: expectedAccount,
        accountKey: expectedAccountKey,
        containerURL: expectedContainerURL,
        settings: {
          autoMkdirOnFileUpload: false
        }
      });
    });

    it("sets them to their default values", () => {
      expect(connection.settings.autoMkdirOnFileUpload).toEqual(false);
    });
  });

  describe("with credentials stored in file", () => {
    const expectedAccount = "expected-account-from-file";
    const expectedURL = "https://not-microsoft.com/";
    const expectedContainer = "myContainerFromFile";
    const expectedContainerURL = `${expectedURL}${expectedContainer}`;

    let connection;

    beforeEach(async () => {
      jest.mock(
        "../../../../credentials/azure_blob.json",
        () => {
          return {
            account: "expected-account-from-file",
            accountKey: "expected-account-from-file-key",
            containerURL: "https://not-microsoft.com/myContainerFromFile"
          };
        },
        { virtual: true }
      );

      jest.spyOn(fs, "existsSync").mockImplementationOnce(() => { return true });

      connection = await integration({});
    });

    it("sets url", () => {
      expect(connection.containerClient.url).toEqual(expectedContainerURL);
    });

    it("sets accountName", () => {
      expect(connection.containerClient.credential.accountName).toEqual(expectedAccount);
    });
  });


  describe("without any parameters", () => {
    it("throws an error", async () => {
      jest.mock(
        "../../../../credentials/azure_blob.json",
        () => {
          return {};
        },
        { virtual: true }
      );

      await expect(integration({})).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});