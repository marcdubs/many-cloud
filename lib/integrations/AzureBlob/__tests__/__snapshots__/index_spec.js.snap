// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Blob without any parameters throws an error 1`] = `"No way to connect to Azure Blob Container provided. Need account + accountKey + containerURL."`;
